src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -113,8 +96,20 @@ class RequestHandlerComponent extends Component
      */
     public function __construct(ComponentRegistry $registry, array $config = [])
     {
+        if (!isset($config['viewClassMap'])) {
+            $config['viewClassMap'] = [
+                'json' => 'Json',
+                'xml' => 'Xml',
+                'ajax' => 'Ajax'
+            ];
+        }
+        if (!isset($config['inputTypeMap'])) {
+            $config['inputTypeMap'] = [
+                'json' => ['json_decode', true],
+                'xml' => [[$this, 'convertXml']],
+            ];
+        }
```

Comment:
Yeah, I realised that was probably the reason after posting the first comment. 

I still feel that this is cleaner code than isset checks:

    $config += ['inputTypeMap' => [...], viewClassMap=> [...]];